client/client.go

```diff
@@ -58,6 +60,14 @@ func Configure(logger hclog.Logger, config interface{}) (schema.ClientMeta, erro
 	serviceAccountKeyJSON := []byte(providerConfig.ServiceAccountKeyJSON)
 	if len(serviceAccountKeyJSON) == 0 {
 		serviceAccountKeyJSON = []byte(os.Getenv(serviceAccountEnvKey))
+		var v map[interface{}]interface{}
+		err := json.Unmarshal(serviceAccountKeyJSON, &v)
+		var syntaxError *json.SyntaxError
+		if errors.As(err, &syntaxError) {
+			return nil, fmt.Errorf("the environment variable %v should contain valid JSON and not just a file path to a JSON file", serviceAccountEnvKey)
```

Comment:
I'm not sure about assuming the value being a file path, could be confusing if it's just an invalid json.

Reply:
Either we can have the error message say `the environment variable %v should contain valid JSON object` or we could check to see if the path exists and if the path exists then we can let the customer know that they shouldn't specify a path

Reply:
Was thinking about that, but also didn't want to overengineer the solution. I thought the most common issue is to supply a path and to state that. I'm fine with either, but when we check that the value is a path we could also just resolve it and feed the content of the path.

Reply:
I am fine with just the generic error

Reply:
@bbernays fixed it.